internal/resources/associate_role/resource_test.go

```diff
@@ -0,0 +1,65 @@
+package associate_role_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/labd/terraform-provider-commercetools/internal/acctest"
+	"github.com/labd/terraform-provider-commercetools/internal/utils"
+)
+
+func TestAssociateRoleResource_Create(t *testing.T) {
+	rn := "commercetools_associate_role.sales_manager_associate_role"
+
+	id := "sales_manager_associate_role"
+	key := "sales_manager_europe_region"
+	name := "Sales Manager - Europe"
+
+	config := testAssociateRoleConfig(id, name, key)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAssociateRoleDestroy,
+		Steps: []resource.TestStep{
```

Comment:
Can we add another test checking if all the updated fields actually get updated?